chips/cc26x0/src/uart.rs

```diff
@@ -214,25 +388,50 @@ impl kernel::hil::uart::UART for UART {
 
         self.disable_interrupts();
         self.set_params(params);
-        self.configure();
+        self.configure(); 
     }
 
     fn transmit(&self, tx_data: &'static mut [u8], tx_len: usize) {
-        if tx_len == 0 {
+        let truncated_len = min(tx_data.len(), tx_len);
```

Comment:
Is this really necessary, why not use `tx_len` and trust that the arguments are correct?

Reply:
I think it's worth including in this case. We tend to do it in other similar places in the kernel.

In general it doesn't hurt to be defensive since the compiler will add bounds checks (which will panic if out of bounds) and will slide them if it can tell the bounds have been checked explicitly.